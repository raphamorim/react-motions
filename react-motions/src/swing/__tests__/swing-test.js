import React from 'react';
import { mount } from 'enzyme';
import Swing from '../swing';

describe('Swing', () => {
  it('should render', () => {
    const wrapper = shallow(<Swing>Swing</Swing>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<Swing duration={3}>Swing</Swing>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<Swing fillMode="both">Swing</Swing>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should receive transformOrigin prop', () => {
    const wrapper = mount(<Swing transformOrigin="center">Swing</Swing>);
    expect(wrapper.props().transformOrigin).toEqual("center");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<Swing infinite>Swing</Swing>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
