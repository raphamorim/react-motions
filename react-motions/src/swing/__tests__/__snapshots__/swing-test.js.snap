// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swing should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Swing
    duration={1}
    fillMode="both"
    infinite={false}
    transformOrigin="top center"
  >
    Swing
  </Swing>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Swing",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "swing",
        "WebkitTransformOrigin": "top center",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "swing",
        "transformOrigin": "top center",
      },
    },
    "ref": null,
    "rendered": "Swing",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Swing",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "swing",
          "WebkitTransformOrigin": "top center",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "swing",
          "transformOrigin": "top center",
        },
      },
      "ref": null,
      "rendered": "Swing",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
